src/main/java/Duke.java

```diff
@@ -1,10 +1,237 @@
+                    + "\u2639" + System.lineSeparator() + LINE);
+        }
+        if (taskDescription.isBlank()) {
+            throw new DukeException(LINE + System.lineSeparator() + "Sir, you haven't given me the name of the task "
+                    + "\u2639" + System.lineSeparator() + LINE);
+        }
+        Todo todo = new Todo(command.substring(TODO_SIZE));
+        tasks.add(todo);
+        printListSummary(tasks.get(tasks.indexOf(todo)));
+    }
+
+    private static void createDeadline(String command) throws DukeException{
+        // Reads two substrings as param: 1. The task description after keyword "deadline"
+        //                                2. The actual deadline after "/by " till end of string
+        String dateOrTime;
+        String taskDescription;
+        int dashStart = command.indexOf("/");
+        try {
+            dateOrTime = command.substring(dashStart + DASH_INDX);
+        } catch (Exception e) {
+            throw new DukeException(LINE + System.lineSeparator() + "Sir, you haven't given me a valid deadline "
+                     + "\u2639" + System.lineSeparator() + LINE);
+        }
+        try {
+            taskDescription = command.substring(DEADLINE_SIZE, dashStart);
+        } catch (Exception e){
+            throw new DukeException(LINE + System.lineSeparator() + "Sir, I'm afraid that command is invalid. " + "\u2639"
+                    + System.lineSeparator() + "Please frame your request in this format: deadline CS2113T Assg /by Wed 2359 hrs"
+                    + System.lineSeparator() + LINE);
+        }
+        Deadline deadline = new Deadline(command.substring(DEADLINE_SIZE, dashStart), dateOrTime);
+        tasks.add(deadline);
+        printListSummary(tasks.get(tasks.indexOf(deadline)));
+    }
+
+    private static void createEvent(String command) throws DukeException {
+        String dateOrTime;
+        String taskDescription;
+        int dashStart = command.indexOf("/");
+        try {
+            dateOrTime = command.substring(dashStart + DASH_INDX);
+        } catch (Exception e) {
+            throw new DukeException(LINE + System.lineSeparator() + "Sir, you haven't given me a valid event date "
+                    + "\u2639" + System.lineSeparator() + LINE);
+        }
+        try {
+            taskDescription = command.substring(EVENT_SIZE, dashStart);
+        } catch (Exception e){
+            throw new DukeException(LINE + System.lineSeparator() + "Sir, I'm afraid that command is invalid." + "\u2639"
+                    + System.lineSeparator() + "Please frame your request in this format: event My B'day /at 23/07/1999"
+                    + System.lineSeparator() + LINE);
+        }
+        Event event = new Event((command.substring(EVENT_SIZE, dashStart)), dateOrTime);
+        tasks.add(event);
+        printListSummary(tasks.get(tasks.indexOf(event)));
+    }
```

Comment:
Consider implementing a TaskManager class to handle all tasks related commands/methods. TaskList can be stored in this class as well